src/Symfony/Component/HttpFoundation/StreamedJsonResponse.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+use Generator;
+
+/**
+ * StreamedJsonResponse represents a streamed HTTP response for JSON.
+ *
+ * A StreamedJsonResponse uses a structure and generics to create an
+ * efficient resource saving JSON response.
+ *
+ * It uses flush after a specified flush size to directly stream the data.
+ *
+ * @see flush()
+ *
+ * @author Alexander Schranz <alexander@sulu.io>
+ *
+ * Example usage:
+ *
+ * $response = new StreamedJsonResponse(
+ *     // json structure with replace identifiers
+ *     [
+ *         '_embedded' => [
+ *             'articles' => '__articles__',
+ *         ],
+ *     ],
+ *     // array of generators with replace identifier used as key
+ *     [
+ *         '__articles__' => (function (): \Generator { // any method or function returning a Generator
+ *              yield ['title' => 'Article 1'];
+ *              yield ['title' => 'Article 2'];
+ *              yield ['title' => 'Article 3'];
+ *         })(),
+ *     ]
+ * );
+ */
+class StreamedJsonResponse extends StreamedResponse
+{
+    public const DEFAULT_ENCODING_OPTIONS = JsonResponse::DEFAULT_ENCODING_OPTIONS;
+
+    protected int $encodingOptions = self::DEFAULT_ENCODING_OPTIONS;
```

Comment:
@stof I would also go that way I just used protected because it was protected in [JsonResponse object](https://github.com/symfony/symfony/blob/4c6049e9b432f511b6b7c32eb495e2dfeeee43e4/src/Symfony/Component/HttpFoundation/JsonResponse.php#L34). But fine to make this private.

Reply:
fixed :)